cg/services/orders/store_order_services/store_metagenome_order.py

```diff
@@ -1,131 +1,114 @@
-        case: Case = self.status_db.get_case_by_name_and_customer(
-            customer=customer, case_name=str(ticket_id)
-        )
-        case_dict: dict = items[0]
-        status_db_order = Order(
-            customer=customer,
-            order_date=datetime.now(),
-            ticket_id=int(ticket_id),
-        )
+        db_order: DbOrder = self._create_db_order(order)
+        priority: PriorityEnum = order.samples[0].priority
+        db_case = self._create_db_case(order=order, customer=customer, priority=priority)
         with self.status_db.session.no_autoflush:
-            for sample in case_dict["samples"]:
-                new_sample = self.status_db.add_sample(
-                    name=sample["name"],
-                    sex=Sex.UNKNOWN,
-                    comment=sample["comment"],
-                    control=sample["control"],
-                    internal_id=sample.get("internal_id"),
-                    order=order,
-                    ordered=ordered,
-                    original_ticket=ticket_id,
-                    priority=sample["priority"],
-                )
-                new_sample.customer: Customer = customer
-                application_tag: str = sample["application"]
-                application_version: ApplicationVersion = (
-                    self.status_db.get_current_application_version_by_tag(tag=application_tag)
-                )
-                if application_version is None:
-                    raise OrderError(f"Invalid application: {sample['application']}")
-                new_sample.application_version: ApplicationVersion = application_version
-                new_samples.append(new_sample)
-
-                if not case:
-                    case = self.status_db.add_case(
-                        data_analysis=Workflow(case_dict["data_analysis"]),
-                        data_delivery=DataDelivery(case_dict["data_delivery"]),
-                        name=str(ticket_id),
-                        panels=None,
-                        priority=case_dict["priority"],
-                        ticket=ticket_id,
-                    )
-                    case.customer = customer
-                    self.status_db.session.add(case)
-                    self.status_db.session.commit()
-
+            for sample in order.samples:
+                db_sample = self._create_db_sample(order=order, sample=sample, customer=customer)
                 new_relationship: CaseSample = self.status_db.relate_sample(
-                    case=case, sample=new_sample, status=StatusEnum.unknown
+                    case=db_case, sample=db_sample, status=StatusEnum.unknown
                 )
                 self.status_db.session.add(new_relationship)
-        status_db_order.cases.append(case)
-        self.status_db.session.add(status_db_order)
+                new_samples.append(db_sample)
+        db_order.cases.append(db_case)
+        self.status_db.session.add(db_order)
```

Comment:
same as in the microbial PR, replace these for the store functions defined in https://github.com/Clinical-Genomics/cg/blob/acfd69475c514ba460dc7ff47e6099d02b3710ea/cg/store/store.py#L31